javagen/src/main/java/com/azure/autorest/model/clientmodel/ClassType.java

```diff
@@ -696,8 +738,48 @@ public Builder prototypeAsDouble() {
+
+            if (!JavaSettings.getInstance().isBranding()) {
+                if (Objects.equals(clazz, com.azure.core.util.ExpandableStringEnum.class)
+                        || Objects.equals(clazz, com.azure.core.util.Context.class)
+                        || Objects.equals(clazz, com.azure.core.util.BinaryData.class)
+                        || Objects.equals(clazz, com.azure.core.util.ClientOptions.class)) {
+                    this.packageName("com.generic.core.models");
+                } else if (Objects.equals(clazz, com.azure.core.http.HttpHeaders.class)
+                        || Objects.equals(clazz, com.azure.core.http.HttpHeader.class)) {
+                    this.packageName("com.generic.core.models");
+                    // HttpHeaders -> Headers
+                    this.name(clazz.getSimpleName().replace("Http", ""));
+                } else if (Objects.equals(clazz, com.azure.core.util.Configuration.class)) {
+                    this.packageName("com.generic.core.util.configuration");
+                } else if (Objects.equals(clazz, com.azure.core.http.policy.RetryOptions.class)) {
+                    this.packageName("com.generic.core.http.policy.retry");
+                } else if (Objects.equals(clazz, com.azure.core.http.rest.RequestOptions.class)
+                        || Objects.equals(clazz, com.azure.core.http.HttpHeaderName.class)) {
+                    this.packageName("com.generic.core.http.models");
+                } else if (Objects.equals(clazz, com.azure.core.http.rest.Response.class)
+                        || Objects.equals(clazz, com.azure.core.http.rest.SimpleResponse.class)
+                        || Objects.equals(clazz, com.azure.core.http.rest.RestProxy.class)) {
+                    this.packageName("com.generic.core.http");
+                } else if (Objects.equals(clazz, com.azure.core.http.HttpClient.class)) {
+                    this.packageName("com.generic.core.http.client");
+                } else if (Objects.equals(clazz, com.azure.core.http.HttpPipeline.class)
+                        || Objects.equals(clazz, com.azure.core.http.HttpPipelineBuilder.class)
+                        || Objects.equals(clazz, com.azure.core.http.policy.HttpPipelinePolicy.class)) {
+                    this.packageName("com.generic.core.http.pipeline");
+                } else if (Objects.equals(clazz, com.azure.core.http.policy.HttpLogOptions.class)
+                        || Objects.equals(clazz, com.azure.core.http.policy.HttpLoggingPolicy.class)) {
+                    this.packageName("com.generic.core.http.policy.logging");
+                } else {
+                    this.packageName(clazz.getPackage().getName()
+                            .replace(Package.AZURE_CORE_PACKAGE_NAME, Package.GENERIC_CORE_PACKAGE_NAME)
+                            .replace(Package.AZURE_JSON_PACKAGE_NAME, Package.GENERIC_JSON_PACKAGE_NAME));
+                }
+            }
```

Comment:
Though we have most generic/azure package/name here, this does not look good.

We might want to be able to write e.g. `Package.CORE.HttpPipeline` (`Package.CORE` would map it to generic or azure) instead.

Though would keep it temporary here, before generic-core get stable.